storage_service/locations/models/package.py

```diff
@@ -522,10 +522,43 @@ def extract_file(self, relative_path='', extract_path=None):
             output_path = os.path.join(extract_path, basename)
 
         if self.is_compressed:
-            command = ['unar', '-force-overwrite', '-o', extract_path, full_path]
+            # The original command used to extract the compressed file at
+            # full_path was::
+            #
+            #     command = ['unar', '-force-overwrite', '-o', extract_path, full_path]
+            #
+            # The problem with this command is that unar treats __MACOSX .rsrc
+            # files differently than 7z does. `7z x ...` is one of the commands
+            # (the typical one?) used in the extractContents_v0.0 client
+            # script. 7z converts these .rsrc files to ._-prefixed files.
+            # Similar behaviour with unar can be achieved by passing `-k
+            # hidden`. However, while a command like::
+            #
+            #     command = ['unar', '-force-overwrite', '-k', 'hidden', '-o', extract_path, full_path]
+            #
+            # preserves the .rsrc MACOSX files as ._-prefixed files, it does so
+            # differently than 7z does: the resulting .-prefixed files have
+            # different sizes than those created via unar. This makes
+            # ``bag.validate`` choke.
+            #
+            # To get FPR commands used for extraction in extractContents_v0.0:
+            # SELECT command
+            #   FROM fpr_fpcommand
+            #     WHERE uuid IN ('a03a3a6d-5695-4800-a16d-edd7143cc438',
+            #                    '8f41dc6f-05eb-46d4-9b22-0a0d74673510',
+            #                    'b8864e2c-87a0-4aec-a8cf-a747a073be28');
+            #
+            # TODOs/Questions:
+            # 1. Will the relative/subpath final argument to 7z work in the
+            #    same way that it does in unar?
```

Comment:
This should be fixed before this can be merged, since this is the same function used to extract single files.